### PR TITLE
[IMP] purchase_order_product_recommendation: extra filters

### PR DESCRIPTION
**Pull request by chienandalu**
- Extract domain build to an independent method for easy extension by
third modules
- Extract product filter to an independent method for easy extension.
- New product internal category filter.
- New "Show all products" filter that allows to easily show all
purchable products.

cc @Tecnativa TT20722

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Review by HviorForgeFlow — APPROVED** [review]
:+1:

**Comment by chienandalu** [comment]
I'll be adding some changes, please don't merge yet

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Fine

**Comment by chienandalu** [comment]
It can be reviewed now :)

**Review by HviorForgeFlow — APPROVED** [review]

File: purchase_order_product_recommendation/wizards/purchase_order_recommendation.py [hunk 1]
```diff
@@ -42,10 +42,20 @@ class PurchaseOrderRecommendation(models.TransientModel):
              'Leave it as 0 to set no limit',
     )
     show_all_partner_products = fields.Boolean(
-        string='Show all products',
+        string='Show all supplier products',
         default=False,
         help='Show all products with supplier infos for this supplier',
```
Comment (HviorForgeFlow) [inline purchase_order_product_recommendation/wizards/purchase_order_recommendation.py:47]:
```suggestion
         help='Show all products with supplier info for this supplier',
```

Comment (HviorForgeFlow) [reply purchase_order_product_recommendation/wizards/purchase_order_recommendation.py:47]: Out of the aim of this PR, If product variant is not match at vendor price, it should display product template price, shouldn't it?

**Comment by pedrobaeza** [comment]
I'm merging this, but if @chienandalu doesn't answer that questions, you can put an issue about that:

/ocabot merge patch

File: purchase_order_product_recommendation/wizards/purchase_order_recommendation.py (hunk 1 above)
Comment (chienandalu) [reply purchase_order_product_recommendation/wizards/purchase_order_recommendation.py:47]: Mmm. I don't think so. It should be 0, as there's no price history with that provider yet and you don't have much control on it anyway.

**Pull request merged**